.github/workflows/server-benchmark.yml

```diff
@@ -24,17 +26,102 @@ jobs:
         key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
         restore-keys: |
           ${{ runner.os }}-gradle-
-    # JDK is needed to generate code
+      # Pinned to the commit hash of v1.3.0
+    - uses: Swatinem/rust-cache@842ef286fff290e445b90b4002cc9807c3669641
+      with:
+        sharedKey: ${{ runner.os }}-${{ env.rust_version }}-${{ github.job }}
+        target-dir: ./target
     - name: Set up JDK
       uses: actions/setup-java@v1
       with:
         java-version: ${{ env.java_version }}
-    # Install Rust
-    - uses: actions-rs/toolchain@v1
+    - name: Install Rust
+      uses: actions-rs/toolchain@v1
       with:
         toolchain: ${{ env.rust_version }}
+        components: ${{ env.rust_toolchain_components }}
         default: true
     - name: Run integration tests
       run: |
         cd rust-runtime/aws-smithy-http-server/examples && \
           make && cargo test
+
+  run-benchmark:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout PR
+      uses: actions/checkout@v3
+      with:
+        path: pull-request
+    - name: Checkout origin/main
+      uses: actions/checkout@v3
+      with:
+        repository: awslabs/smithy-rs
+        path: origin-main
+        # ref: main
+        ref: ${{ github.event.pull_request.head.sha }}
+    - uses: actions/cache@v2
+      name: Gradle Cache
+      with:
+        path: |
+          ~/.gradle/caches
+          ~/.gradle/wrapper
+        key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+        restore-keys: |
+          ${{ runner.os }}-gradle-
+      # Pinned to the commit hash of v1.3.0
+    - name: Rust Cache
+      uses: Swatinem/rust-cache@842ef286fff290e445b90b4002cc9807c3669641
+      with:
+        sharedKey: ${{ runner.os }}-${{ env.rust_version }}-${{ github.job }}
+        target-dir: ./target
+    - name: Set up JDK
+      uses: actions/setup-java@v1
+      with:
+        java-version: ${{ env.java_version }}
+    - name: Install Rust
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: ${{ env.rust_version }}
+        components: ${{ env.rust_toolchain_components }}
+        default: true
+    - name: Install benchmarks dependencies
+      run: sudo apt-get install -y ${{ env.apt_dependencies }}
+    # Ubuntu 20.04 doesn't have wrk packaged, hence we need to build it 🤦
+    # This will go away as soon as GitHub supports Ubuntu 21.04.
```

Comment:
22.04 LTS? It's not in 21.04, only in 21.10: https://packages.ubuntu.com/impish/wrk

Reply:
I'll fix this.